ci/cifailurejob.yaml

```diff
@@ -0,0 +1,166 @@
+                    name: 'ccp-pipeline-ci',
+                    label: 'ccp-pipeline-ci',
+                    serviceAccount: 'jenkins',
+                    containers: [
+                      containerTemplate(
+                        name: 'jnlp',
+                        image: 'registry.centos.org/pipeline-images/ccp-openshift-slave',
+                        ttyEnabled: true,
+                        alwaysPullImage: true,
+                        workingDir: '/tmp',
+                        privileged: true,
+                        args: '${computer.jnlpmac} ${computer.name}'
+                      )
+                    ],
+                    volumes: [
+                      hostPathVolume(
+                        hostPath: '/var/run/docker.sock',
+                        mountPath: '/var/run/docker.sock'
+                      )
+                    ]
+                )
+                {
+                    node('ccp-pipeline-ci') {
+                        def failure_build_id = "failure_build_id"
+                        stage("Trigger builds"){
+                            failure_build_id = sh(returnStdout: true, script: "oc start-build nshaikh-build-fail-test-latest -n cccp|cut -f 2 -d ' '").trim()
+                            echo "Failure check build id: ${failure_build_id}"
+                        }
+                        stage("Wait for build to start")
+                        {
+                            waitForBuildToStart(failure_build_id)
+                        }
+                        stage("Check checkout status")
+                        {
+                            checkStageStatus(failure_build_id, 0, "success")
+                        }
+                        stage("Check pre-build status")
+                        {
+                            checkStageStatus(failure_build_id, 1, "success")
+                        }
+                        stage("Check Dockerfile linter status")
+                        {
+                            checkStageStatus(failure_build_id, 2, "success")
+                        }
+                        stage("Check image build status")
+                        {
+                            checkStageStatus(failure_build_id, 3, "failed")
+                        }
+                        stage("Check image scan status")
+                        {
+                            checkStageStatus(failure_build_id, 4, "not_available")
+                        }
+                        stage("Check image delivery status")
+                        {
+                            checkStageStatus(failure_build_id, 5, "not_available")
+                        }
+                        stage("Check image cleanup status")
+                        {
+                            checkStageStatus(failure_build_id, 6, "not_available")
+                        }
+
+                    }
+                }
+                def waitForBuildToStart(build_id)
```

Comment:
Both the builds were in the same file running in parallel, now we have separated it out in two files. But I could not make it as library for now. I intend to use them as library functions so that we can stop repeating code and if there is any update it gets applied to both the pipelines. lets keep it as it is now, we will use library functions in next iteration.